src/fs_report.rs

```diff
@@ -0,0 +1,147 @@
+    });
+    Ok(())
+}
+
+// Writes each of the provided `ingredient`'s thumbnails, if present, to the `destination` path.
+fn write_ingredient_thumbnails(ingredients: &[Ingredient], destination: &Path) -> Result<()> {
+    ingredients
+        .iter()
+        .filter_map(|i| {
+            i.thumbnail().map(|(format, bytes)| {
+                let title = add_extension(i.title(), format);
+                (title, bytes)
+            })
+        })
+        .map(|(title, bytes)| File::create(destination.join(title))?.write_all(bytes))
+        .collect::<Result<Vec<()>, io::Error>>()?;
+    Ok(())
+}
+
+/// Writes the report of the manifest, including the manifest's thumbnails and ingredient thumbnails,
+/// to the provided `destination_path`.
+pub(crate) fn write_report_for_path(
+    manifest_path: &Path,
+    destination_path: &Path,
+    is_detailed: bool,
+) -> Result<()> {
+    let store = ManifestStore::from_file(manifest_path)?;
+    create_dir_all(destination_path)?;
+
+    store
+        .manifests()
+        .iter()
+        .enumerate()
+        .map(|(i, (_, manifest))| {
+            let label = manifest
+                .label()
+                .map(str::to_string)
+                .unwrap_or_else(|| format!("manifest_{}", i))
+                .replace(['/', ':'], "-");
+
+            let claim_path = destination_path.join(label);
+            create_dir_all(&claim_path)?;
+
+            write_manifest_thumbnail(manifest, &claim_path)
+                .and_then(|_| write_ingredient_thumbnails(manifest.ingredients(), &claim_path))
+        })
+        .collect::<Result<Vec<()>>>()?;
+
+    let manifest_bytes = match is_detailed {
+        true => ManifestStoreReport::from_file(manifest_path)?.to_string(),
+        false => store.to_string(),
+    }
+    .into_bytes();
+
+    File::create(destination_path.join("manifest.json"))?.write_all(&manifest_bytes)?;
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::fs_report::add_extension;
+
+    #[test]
+    fn test_file_extension() {
+        assert_eq!(add_extension("filename.png", "image/png"), "filename.png");
+    }
+
+    #[test]
+    fn test_file_extension_not_equal_ingredient_name() {
+        assert_eq!(
+            add_extension("filename.psd", "image/png"),
+            "filename.psd.png"
+        );
+    }
+
+    #[test]
+    fn test_file_extension_not_equal_ingredient_name_and_known_type() {
+        assert_eq!(
+            add_extension("filename.png", "image/jpg"),
+            "filename.png.jpg"
+        );
+    }
+
+    #[test]
+    fn test_file_extension_ingredient_has_no_extension() {
+        assert_eq!(add_extension("filename", "image/jpeg"), "filename.jpg");
+    }
+
+    #[test]
+    fn test_file_extension_ingredient_filename_has_period_in_name() {
+        assert_eq!(
+            add_extension("filename.final.version", "image/jpeg"),
+            "filename.final.version.jpg"
+        );
+    }
```

Comment:
Hey, @gpeacock: I went back and forth on the behavior for adding the correct extension to the ingredient title because of the edge-case in this test. We could ignore the edge-case and just replace whatever comes after the last period. It'll definitely make the code cleaner!

```rust
let mut path = PathBuf::from(&file_name);
path.set_extension(&format_extension);
```

vs 

All of this: 
https://github.com/contentauth/c2patool/pull/91/files#diff-9c12fef2a4d35ee776937bc6bae5c05828d27811fc3129aad5dd78c7bc5c285fR23-R45

Reply:
I think that is ok for this and it does make things much cleaner